packit/cli/builds/koji_build.py

```diff
@@ -13,10 +14,54 @@
 from packit.exceptions import PackitCommandFailedError, ensure_str
 from packit.exceptions import PackitConfigException
 from packit.utils.changelog_helper import ChangelogHelper
+from packit.config import get_local_package_config
+from packit.config.common_package_config import MultiplePackages
 
 logger = logging.getLogger(__name__)
 
 
+def iterate_packages(func):
+    """
+    Decorator for dealing with packages in a package configuration
+
+    * if packages are specified as an option in CLI then
+      do a koji build just for them
+    * if packages are not specified as an option in CLI but
+      there are multiple packages in the configuration
+      then do a koji build for all of them
+    * if there is just one package in the configuration
+      then do a single koji build
+
+    This method (iterate_packages) **has not** "package_config"
+    in its kwargs but calls a method (func) who needs a
+    package_config in its args!
+    """
+
+    @functools.wraps(func)
+    def covered_func(*args, **kwargs):
+        path_or_url = kwargs["path_or_url"]
+        config = kwargs["config"]
+        packages_config: MultiplePackages = get_local_package_config(
+            path_or_url.working_dir,
+            repo_name=path_or_url.repo_name,
+            try_local_dir_last=True,
+            package_config_path=config.package_config_path,
+        )
+        if "packages" in kwargs and kwargs["packages"]:
+            for package in kwargs["packages"].split(","):
+                kwargs["package_config"] = packages_config.packages[package]
+                func(*args, **kwargs)
+        elif hasattr(packages_config, "packages"):
+            for _, package_config in packages_config.packages.items():
+                kwargs["package_config"] = package_config
+                func(*args, **kwargs)
+        else:
+            kwargs["package_config"] = packages_config
+            func(*args, **kwargs)
+
+    return covered_func
```

Comment:
this looks like an efficient solution here for the CLI though I wonder how we could apply it in packit-service too.

Reply:
In this PR https://github.com/packit/packit-service/pull/1873 my considerations about `packit-service`.